serde/src/main/java/com/amazon/ionhiveserde/formats/IonInputFormat.java

```diff
@@ -116,19 +135,25 @@ public BytesWritable createValue() {
             return new BytesWritable();
         }
 
+        // CompressionInputStream and FSDataInputStream are both Seekable, but have no common
+        // Seekable parents. We can confidently cast our InputStream to Seekable.
         @Override
         public final long getPos() throws IOException {
-            return fsDataInputStream.getPos();
+            return ((Seekable)in).getPos();
         }
 
         @Override
         public final void close() throws IOException {
-            this.fsDataInputStream.close();
+            this.in.close();
         }
 
         @Override
+        // We don't know why `progress` is equal to 0 instead of 1 when start == end, what Hadoop is doing since long
+        // times ago is setting it to 0.
```

Comment:
Just for fun let's link to the source of these behaviors.

The original split support issue: https://issues.apache.org/jira/browse/HADOOP-451
The original patch that introduced this:  https://svn.apache.org/viewvc/lucene/hadoop/trunk/src/java/org/apache/hadoop/mapred/TextInputFormat.java?r1=469596&r2=488438&pathrev=502021&diff_format=h